sam-react-app/coreconcept/src/counter/Counter.tsx

```diff
@@ -0,0 +1,75 @@
+import React, {Component, createElement} from "react";
+
+class Counter extends Component {
+    state = {
+        counter: 0,
+    };
+    handleOnClick = (event: Event) => {
+        const element = event?.target as HTMLElement;
+        const componentId = element.getAttribute('id');
+        switch(componentId){
+            case 'increment':
+                this.setState({
+                    counter: this.state.counter + 1,
+                });
+                break;
+            case 'decrement':
+                this.setState({
+                    counter: this.state.counter - 1,
+                });
+                break;
+            default:
+                throw Error('Unknown action');
+        }
+    }
+    handleInputChange = (event: Event) => {
+        const inputElement = event?.target as HTMLInputElement;
+        this.setState({
+            counter: Number(inputElement.value),
+        });
```

Comment:
```suggestion
        this.setState({
            counter: Number(event.target.value),
        });
```
This is a very common pattern in react, you can simplify it. See examples [in the docs](https://legacy.reactjs.org/docs/forms.html).

Reply:
done